sbtc-core/src/signer/mod.rs

```diff
@@ -0,0 +1,169 @@
+/// An sBTC signer configuration
+pub mod config;
+
+use std::collections::HashMap;
+
+use crate::SBTCError;
+use crate::{signer::config::Config, SBTCResult};
+use bitcoin::{Address, Network, PrivateKey, PublicKey, Transaction as BitcoinTransaction};
+use p256k1::ecdsa;
+use url::Url;
+use wsts::{bip340::SchnorrProof, common::Signature};
+
+/// A Stacks transaction
+/// TODO: replace with the core library's StacksTransaction
+pub struct StacksTransaction {}
+
+/// An sBTC transaction
+/// TODO: replace with the core library's SBTCTransaction
+/// This could be a BTC transaction or a STX transaction
+/// depending on https://github.com/Trust-Machines/stacks-sbtc/pull/595
+pub struct SBTCTransaction {}
+
+#[derive(Default, Clone, Debug)]
+/// Signers' public keys required for weighted distributed signing
+pub struct PublicKeys {
+    /// Signer ids to public key mapping
+    pub signer_ids: HashMap<u32, ecdsa::PublicKey>,
+    /// Vote ids to public key mapping
+    pub vote_ids: HashMap<u32, ecdsa::PublicKey>,
+}
+
+/// sBTC Keys trait for retrieving signer IDs, vote IDs, and public keys
+trait Keys {
+    fn public_keys(&self) -> SBTCResult<PublicKeys>;
+    fn coordinator_public_key(&self, tx: &BitcoinTransaction) -> SBTCResult<PublicKey>;
```

Comment:
Should we have this method return an ordered list of public keys, the first one being the primary coordinator, second one being the secondary etc.?